geoh5py/objects/object_base.py

```diff
@@ -136,26 +163,25 @@ def add_data(
                 )
 
             attr["name"] = name
+
             attr, validate_property_group = self.validate_association(
                 attr, property_group=property_group, **kwargs
             )
 
             entity_type = self.workspace.validate_data_type(attr, attr.get("values"))
 
-            kwargs = {"parent": self, "association": attr["association"]}
+            kwargs = {"parent": self}
             for key, val in attr.items():
-                if key in ["parent", "association", "entity_type", "type"]:
-                    continue
-
-                kwargs[key] = val
+                if key not in ["parent", "entity_type", "type"]:
+                    kwargs[key] = val
 
             data_object = self.workspace.create_entity(
-                Data, entity=kwargs, entity_type=entity_type, compression=compression
+                Data,  # type: ignore
```

Comment:
Better to keep the typing on workspace create_entity then a blind type ignore

Reply:
it's not a type:ignore, it's because Data is an abstact class. 
The typing out of this function is "Data", but because it's an abstract class, mypy don't like that... 
I'll find a way to make it better.